### ISSUE TITLE
TinyXML build failure

### ISSUE THREAD

**Issue by seccpur** [microsoft/vcpkg#21873]
**Host Environment**
- OS: windows 10
- Compiler: VC++ 2022

**To Reproduce**
`./vcpkg install \tinyxml:x64-windows`

**Failure logs** 
```
PS D:\Repo\vcpkg-master> .\vcpkg install tinyxml:x64-windows
Computing installation plan...
A suitable version of cmake was not found (required v3.21.1). Downloading portable cmake v3.21.1...
Downloading cmake...
  https://github.com/Kitware/CMake/releases/download/v3.21.1/cmake-3.21.1-windows-i386.zip -> D:\Repo\vcpkg-master\downloads\cmake-3.21.1-windows-i386.zip
Extracting cmake...
A suitable version of 7zip was not found (required v19.0.0). Downloading portable 7zip v19.0.0...
Downloading 7zip...
  https://www.7-zip.org/a/7z1900-x64.msi -> D:\Repo\vcpkg-master\downloads\7z1900-x64.msi
Extracting 7zip...
The following packages will be built and installed:
    tinyxml[core]:x64-windows -> 2.6.2#8
Detecting compiler hash for triplet x64-windows...
A suitable version of git was not found (required v2.32.0). Downloading portable git v2.32.0...
Downloading git...
  https://github.com/git-for-windows/git/releases/download/v2.32.0.windows.2/PortableGit-2.32.0.2-32-bit.7z.exe -> D:\Repo\vcpkg-master\downloads\PortableGit-2.32.0.2-32-bit.7z.exe
Extracting git...
A suitable version of powershell-core was not found (required v7.2.0). Downloading portable powershell-core v7.2.0...
Downloading powershell-core...
  https://github.com/PowerShell/PowerShell/releases/download/v7.2.0/PowerShell-7.2.0-win-x86.zip -> D:\Repo\vcpkg-master\downloads\PowerShell-7.2.0-win-x86.zip
Extracting powershell-core...
Restored 0 packages from C:\Users\PROJECT\AppData\Local\vcpkg\archives in 132.9 us. Use --debug to see more details.
Starting package 1/1: tinyxml:x64-windows
Building package tinyxml[core]:x64-windows...
-- Note: tinyxml only supports static library linkage. Building static library.
-- Downloading https://sourceforge.net/projects/tinyxml/files/tinyxml/2.6.2/tinyxml_2_6_2.tar.gz/download;https://sourceforge.net/projects/tinyxml/files/tinyxml/2.6.2/tinyxml_2_6_2.tar.gz/download?use_mirror=cfhcable;https://sourceforge.net/projects/tinyxml/files/tinyxml/2.6.2/tinyxml_2_6_2.tar.gz/download?use_mirror=pilotfiber;https://sourceforge.net/projects/tinyxml/files/tinyxml/2.6.2/tinyxml_2_6_2.tar.gz/download?use_mirror=gigenet;https://sourceforge.net/projects/tinyxml/files/tinyxml/2.6.2/tinyxml_2_6_2.tar.gz/download?use_mirror=versaweb;https://sourceforge.net/projects/tinyxml/files/tinyxml/2.6.2/tinyxml_2_6_2.tar.gz/download?use_mirror=ayera;https://sourceforge.net/projects/tinyxml/files/tinyxml/2.6.2/tinyxml_2_6_2.tar.gz/download?use_mirror=netactuate;https://sourceforge.net/projects/tinyxml/files/tinyxml/2.6.2/tinyxml_2_6_2.tar.gz/download?use_mirror=phoenixnap;https://sourceforge.net/projects/tinyxml/files/tinyxml/2.6.2/tinyxml_2_6_2.tar.gz/download?use_mirror=astuteinternet;https://sourceforge.net/projects/tinyxml/files/tinyxml/2.6.2/tinyxml_2_6_2.tar.gz/download?use_mirror=freefr;https://sourceforge.net/projects/tinyxml/files/tinyxml/2.6.2/tinyxml_2_6_2.tar.gz/download?use_mirror=netcologne;https://sourceforge.net/projects/tinyxml/files/tinyxml/2.6.2/tinyxml_2_6_2.tar.gz/download?use_mirror=deac-riga;https://sourceforge.net/projects/tinyxml/files/tinyxml/2.6.2/tinyxml_2_6_2.tar.gz/download?use_mirror=excellmedia;https://sourceforge.net/projects/tinyxml/files/tinyxml/2.6.2/tinyxml_2_6_2.tar.gz/download?use_mirror=iweb;https://sourceforge.net/projects/tinyxml/files/tinyxml/2.6.2/tinyxml_2_6_2.tar.gz/download?use_mirror=jaist;https://sourceforge.net/projects/tinyxml/files/tinyxml/2.6.2/tinyxml_2_6_2.tar.gz/download?use_mirror=jztkft;https://sourceforge.net/projects/tinyxml/files/tinyxml/2.6.2/tinyxml_2_6_2.tar.gz/download?use_mirror=managedway;https://sourceforge.net/projects/tinyxml/files/tinyxml/2.6.2/tinyxml_2_6_2.tar.gz/download?use_mirror=nchc;https://sourceforge.net/projects/tinyxml/files/tinyxml/2.6.2/tinyxml_2_6_2.tar.gz/download?use_mirror=netix;https://sourceforge.net/projects/tinyxml/files/tinyxml/2.6.2/tinyxml_2_6_2.tar.gz/download?use_mirror=ufpr;https://sourceforge.net/projects/tinyxml/files/tinyxml/2.6.2/tinyxml_2_6_2.tar.gz/download?use_mirror=tenet -> tinyxml_2_6_2.tar.gz...
-- Extracting source D:/Repo/vcpkg-master/downloads/tinyxml_2_6_2.tar.gz
-- Applying patch 0001_use_stl.patch
-- Applying patch 0002_export_tinyxml.patch
-- Using source at D:/Repo/vcpkg-master/buildtrees/tinyxml/src/133b5db061-90b87b703c.clean
-- Found external ninja('1.10.2').
-- Configuring x64-windows
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:127 (message):
    Command failed: ninja -v
    Working Directory: D:/Repo/vcpkg-master/buildtrees/tinyxml/x64-windows-rel/vcpkg-parallel-configure
    Error code: 1
    See logs for more information:
      D:\Repo\vcpkg-master\buildtrees\tinyxml\config-x64-windows-out.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_configure_cmake.cmake:345 (vcpkg_execute_required_process)
  ports/tinyxml/portfile.cmake:16 (vcpkg_configure_cmake)
  scripts/ports.cmake:142 (include)


Error: Building package tinyxml:x64-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `git pull` and `.\vcpkg update`.
Then check for known issues at:
  https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+tinyxml
You can submit a new issue at:
  https://github.com/microsoft/vcpkg/issues/new?template=report-package-build-failure.md&title=[tinyxml]+Build+error
including:
  package: tinyxml[core]:x64-windows -> 2.6.2#8
    vcpkg-tool version: 2021-11-24-48b94a6946b8a70abd21529218927fd478d02b6c
    vcpkg-scripts version: unknown

Additionally, attach any relevant sections from the log files above.
```
**Additional context**
Log file:
[config-x64-windows-out.log](https://github.com/microsoft/vcpkg/files/7658498/config-x64-windows-out.log)

**Comment by JackBoosY** [comment]
See comments in https://github.com/microsoft/vcpkg/issues/21580.

We do need to improve `detect-compiler`'s error message.

**Comment by seccpur** [comment]
The matter has been resolved after referring to this link[#21580](https://github.com/microsoft/vcpkg/issues/21580)
